Task10_Auth/src/main/java/AuthSingleton.java

```diff
@@ -0,0 +1,73 @@
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.message.BasicNameValuePair;
+import org.apache.http.util.EntityUtils;
+import reseponsesDTO.TokenDto;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.util.ArrayList;
+import java.util.Properties;
+
+public class AuthSingleton {
+    private static AuthSingleton object;
+    private static String writeToken;
+    private static String readToken;
+    private Properties properties = new LoadProperties().loadProperties();
+
+    private AuthSingleton() {
+        readToken = getToken(properties.getProperty("readAccess"));
+        writeToken = getToken(properties.getProperty("writeAccess"));
+    }
+
+    public static AuthSingleton getInstance() {
+        if (object == null) {
+            object = new AuthSingleton();
+        }
+
+        return object;
+    }
+
+    public String getReadToken() {
+        return "Bearer " + readToken;
+    }
+
+
+    public String getWriteToken() {
+        return "Bearer " + writeToken;
+    }
+
+    private String getToken(String accessType) {
```

Comment:
It makes sense to return TokenDto or just rename this method to getAccessToken f.e.

Reply:
Changed to return TokenDto object